### PR TITLE
Add types for "raygun" package

### PR DESCRIPTION
**Pull request by UberMouse**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

This PR adds typings for the Raygun node package (called raygun). I work at Raygun but I haven't done much work on our Raygun4Node library but the types are correct to the best of my knowledge.

---
This change is [Reviewable](https://reviewable.io/reviews/definitelytyped/definitelytyped/30180)

### CODE REVIEW

File: types/raygun/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            userCustomData?: KeyValueObject;
+            tags: string[];
+            request?: RaygunRequest;
+            user?: RaygunUser | { identifier: string }
+            version?: string;
+        };
+    }
+
+    interface RaygunOfflineStorageProvider<TTransportItem = RaygunPayload, TStorageItem = string> {
+        init(options: any): RaygunOfflineStorageProvider;
```
Comment (SimonSchick) [inline types/raygun/index.d.ts]: Nit, avoid `any`.

File: types/raygun/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    class Client {
+        init(options: RaygunOptions): Client;
+        setUser(user: RaygunUser): Client;
+        setVersion(version: string): Client;
+        onBeforeSend(callback: OnBeforeSend): Client;
+        groupingKey(groupingKey: string): Client;
+        offline(): Client;
+        online(): Client;
+        send(exception: Error | string | object, customData?: KeyValueObject, offlineStorageCallback?: (error?: Error) => void, request?: RaygunRequest, tags?: ReadonlyArray<string>): RaygunPayload;
+        expressHandler(error: Error, request: RaygunRequest, res: any, next: any): void;
```
Comment (SimonSchick) [inline types/raygun/index.d.ts]: You can probably use the default node types here, otherwise use `object` to at least narrow the type.

**Review by SimonSchick — APPROVED** [review]

File: types/raygun/index.d.ts (hunk 1 above)
Comment (UberMouse) [reply types/raygun/index.d.ts]: Should I use `object` instead then? Have to cater for customers providing their own offline storage provider which could take anything as options

File: types/raygun/index.d.ts (hunk 2 above)
Comment (UberMouse) [reply types/raygun/index.d.ts]: Are the express types part of node? This method isn't designed to be consumed from your code, it's just passed straight to express.

File: types/raygun/index.d.ts (hunk 1 above)
Comment (SimonSchick) [reply types/raygun/index.d.ts]: Could make it generic eg. `RaygunOfflineStorageProvider<TTransportItem = RaygunPayload, TStorageItem = string, TOptions = void>`, although that might lead to having to bubble up the generic...

Comment (DanielRosenwasser) [reply types/raygun/index.d.ts]: `any` (or `unknown`) is fine if it really is anything

File: types/raygun/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    }
+}
+
+export default raygun;
```
Comment (DanielRosenwasser) [inline types/raygun/index.d.ts]: This package was declared using `exports.Client = ...`. It should be declared with an `export = ...` like one of the following:

```ts
namespace foo {
  // ...
}

export = foo
```

To import the entity using a default import, you will need to enable `esModuleInterop`.

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/raygun/index.d.ts (hunk 3 above)
Comment (UberMouse) [reply types/raygun/index.d.ts]: I think I've done that @DanielRosenwasser, can you please recheck? Thanks.

**Comment by DanielRosenwasser** [comment]
Thank you!

**Pull request merged**